tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/CounterexampleWriter.scala

```diff
@@ -52,59 +52,69 @@ object CounterexampleWriter {
     val pretty = new PrettyWriter(writer)
 
-    writer.println("%s MODULE counterexample %s\n".format("-" * 25, "-" * 25))
-    writer.println("EXTENDS %s\n".format(rootModule.name))
+    pretty.prettyWriteDoc(pretty.moduleNameDoc("counterexample", 25))
+    pretty.writeln()
+    pretty.prettyWriteDoc(pretty.moduleExtendsDoc(rootModule.name))
+    pretty.writeln()
 
     states.zipWithIndex.foreach {
       case (state, 0) =>
-        writer.println("(* Constant initialization state *)")
-        writer.println(s"\nStateConst ==")
+        pretty.writeln("(* Constant initialization state *)")
+        pretty.writeln(s"\nConstInit ==")
         printStateFormula(pretty, state._2)
-        writer.println()
+        pretty.writeln()
       case (state, j) =>
-        // start counting from 0, for compatibility with TLC counterexamples
+        // Index 0 is reserved for ConstInit, but users expect State0 to
+        // be the initial state, so we shift indices by 1 for print-output
         val i = j - 1
         if (i == 0) {
-          writer.println("(* Initial state *)")
+          pretty.writeln("(* Initial state *)")
         } else {
-          writer.println(s"(* Transition ${state._1} to State$i *)")
+          pretty.writeln(s"(* Transition ${state._1} to State$i *)")
         }
-        writer.println(s"\nState$i ==")
+        pretty.writeln(s"\nState$i ==")
         printStateFormula(pretty, state._2)
-        writer.println()
+        pretty.writeln()
     }
-    writer.print(s"""(* The following formula holds true in the last state and violates the invariant *)
+    pretty.write(s"""(* The following formula holds true in the last state and violates the invariant *)
                     |
                     |""".stripMargin)
     pretty.write(TlaOperDecl("InvariantViolation", List(), notInvariant))
 
-    writer.println("\n\n%s".format("=" * 80))
-    writer.println("\\* Created by Apalache on %s".format(Calendar.getInstance().getTime))
-    writer.println("\\* https://github.com/informalsystems/apalache")
-    writer.close()
+    pretty.writeln()
+    pretty.writeln()
+    pretty.prettyWriteDoc(pretty.moduleTerminalDoc(80))
+    pretty.writeln("\\* Created by Apalache on %s".format(Calendar.getInstance().getTime))
+    pretty.writeln("\\* https://github.com/informalsystems/apalache")
+    pretty.close()
   }
 }
 
 class TlcCounterexampleWriter(writer: PrintWriter) extends TlaCounterexampleWriter(writer) {
-
   override def write(rootModule: TlaModule, notInvariant: NotInvariant, states: List[NextState]): Unit = {
-    writer.print(s"""@!@!@STARTMSG 2262:0 @!@!@
+    // `states` must always contain at least 1 state: the constant initialization
+    // This makes `states.tail` safe, since we have a nonempty sequence
+    assert(states.nonEmpty)
+
+    val pretty = new PrettyWriter(writer)
+
+    pretty.write(s"""@!@!@STARTMSG 2262:0 @!@!@
```

Comment:
In case of TLC, I would rather use a `PrintWriter`, not `PrettyWriter`. The TLC format does not have to be pretty printed.

Reply:
In the original code, TLC writer was already using prettyprinting, in `printStateFormula(new PrettyWriter(writer), state._2)`

Reply:
Then let's not use `PrettyWriter` for TLC counterexamples. TLC counterexamples obviously enforce their own formatting, which we have to beatify later.

Reply:
Just to copy our Slack discussion here:
We cannot get rid of a `PrettyWriter` call here, because `PrintWriter` does not know how to display TLA+ code.
Possible TODO in the future, but not as part of this PR.